Lib/socket.py

```diff
@@ -728,6 +728,144 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
     else:
         raise error("getaddrinfo returns an empty list")
 
+def supports_hybrid_ipv46():
+    """Return True if the platform supports creating a SOCK_STREAM socket
+    which can handle both AF_INET and AF_INET6 (IPv4 / IPv6) connections.
+    """
+    if not has_ipv6 \
+            or not hasattr(_socket, 'IPPROTO_IPV6') \
+            or not hasattr(_socket, 'IPV6_V6ONLY'):
+        return False
+    try:
+        with socket(AF_INET6, SOCK_STREAM) as sock:
```

Comment:
This check can be done once when this function is called for the first time and then cached, right?

Reply:
It can but IMO it's not worth it because:
1) this is a fast operation already: just instantiate a socket, set an option on it, close() it
2) in a server application this is typically supposed to be used only once
...also in case there are systems where `IPV6_V6ONLY` can be set globally (and I don't know if there are) we want a fresh result

Reply:
Linux has `/proc/sys/net/ipv6/bindv6only` which can be changed at anytime...  http://man7.org/linux/man-pages/man7/ipv6.7.html

Reply:
Thanks. I will try experimenting with that to see it doesn't introduce issues or further differences.

Reply:
OK, this seems to determine the default value of `IPV6_V6ONLY` option for newly created sockets:

```
$ echo 0 > /proc/sys/net/ipv6/bindv6only
$ ./python -c "from socket import *; s = socket(AF_INET6); print(s.getsockopt(IPPROTO_IPV6, IPV6_V6ONLY))"
0
$ echo 1 > /proc/sys/net/ipv6/bindv6only
$ ./python -c "from socket import *; s = socket(AF_INET6); print(s.getsockopt(IPPROTO_IPV6, IPV6_V6ONLY))"
1
```

Even if `/proc/sys/net/ipv6/bindv6only` is set to 1, I am able to set it to 0 from Python, and unit-tests for `create_server` will pass. Basically this means that `/proc/sys/net/ipv6/bindv6only` does not disable the functionality. It just determines its default.

Reply:
oh good. :)